### PR TITLE
Update keycloak auth docs to specify how to get jwtRolesPath

### PR DESCRIPTION
**Pull request by averyniceday**
Had some issues when deploying GENIE with oauth2 DAT method. 

Ended up realizing it was an issue with jwtRolesPath. Thought it would help to have clearer docs explaining how to get the path out of the JWT. 

# Checks
- [ ] Runs on heroku
- [ ] Has tests or has a separate issue that describes the types of test that should be created. If no test is included it should explicitly be mentioned in the PR why there is no test.
- [ ] The commit log is comprehensible. It follows [7 rules of great commit messages](http://chris.beams.io/posts/git-commit/). For most PRs a single commit should suffice, in some cases multiple topical commits can be useful. During review it is ok to see tiny commits (e.g. Fix reviewer comments), but right before the code gets merged to master or rc branch, any such commits should be squashed since they are useless to the other developers. Definitely avoid [merge commits, use rebase instead.](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)
- [ ] Is this PR adding logic based on one or more **clinical** attributes? If yes, please make sure validation for this attribute is also present in the data validation / data loading layers (in backend repo) and documented in [File-Formats Clinical data section](https://github.com/cBioPortal/cbioportal/blob/master/docs/File-Formats.md#clinical-data)!

# Any screenshots or GIFs?
If this is a new visual feature please add a before/after screenshot or gif
here with e.g. [Giphy CAPTURE](https://giphy.com/apps/giphycapture) or [Peek](https://github.com/phw/peek)

# Notify reviewers
Read our [Pull request merging
policy](../CONTRIBUTING.md#pull-request-merging-policy). It can help to figure out who worked on the
file before you. Please use `git blame <filename>` to determine that
and notify them either through slack or by assigning them as a reviewer on the PR

### CODE REVIEW

**Review by pvannierop — COMMENTED** [review]
Correct me if I am wrong. Normally, when the user follows the standard installation instructions, the default value (_resource_access::cbioportal::roles_) would be ok, not not? If so, I think pointing the user to this while following standard setup instructions, is not the best place. I would propose to move your instruction to a new section before `### Troubleshooting` while explaining that this is only needed when deviating from default installation config.

What do you think?

**Comment by averyniceday** [comment]
@pvannierop good idea, I'll move it to a troubleshooting section. 👍 Will try to update by early next week and ping you.

**Comment by averyniceday** [comment]
@pvannierop updated the PR and moved the new stuff into the Troubleshooting section. Lmk if there's anything else! :)

File: docs/Authenticating-and-Authorizing-Users-via-keycloak.md [hunk 1]
```diff
@@ -357,3 +357,32 @@ log4j.logger.org.springframework.security=DEBUG
 ```
 
 Then, rebuild the WAR, redeploy, and try to authenticate again.  Your log file will then include hundreds of SAML-specific messages, even the full XML of each SAML message, and this should help you debug the error.
+
+#### Determining jwtRolesPath for OAuth2 Token
+If the generated token returns a 4XX error when used, confirm that the `jwtRolesPath` property matches the roles path.
```
Comment (pvannierop) [inline docs/Authenticating-and-Authorizing-Users-via-keycloak.md:362]: The trigger for looking at this section should not be a 4xx error. I propose to remove this line and replace it for something like this:

> By default user-roles are extracted from path `resource_access::cbioportal::roles` in the JWT json. When the configuration of roles at the realm and client level in a Keycloak instance differ from that described in this manual, configure the path can be specified using the `dat.oauth2.jwtRolesPath` property in the _portal.properties_ file.

**Comment by averyniceday** [comment]
updated

**Review by pvannierop — APPROVED** [review]

**Comment by pvannierop** [comment]
@averyniceday Approved!

**Pull request merged**